demo.py

```diff
@@ -25,115 +27,274 @@
-                    node_color='#00b4d9', font_color='w')
-
-# Save image
-plt.savefig("map.png")
+def read_in_args():
+    """ Read in user specified parameters or use defaults."""
+
+    # Read in user option for random seed
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-s", "--seed", help="set a random seed for scenario", type=int)
+    parser.add_argument("-x", "--width", help="set the width of the grid", type=int)
+    parser.add_argument("-y", "--height", help="set the height of the grid", type=int)
+    parser.add_argument("-p", "--poi", help="set the number of POIs", type=int)
+    parser.add_argument("-c", "--chargers", help="set the number of existing chargers", type=int)
+    parser.add_argument("-n", "--new_chargers", help="set the number of new chargers", type=int)
+    args = parser.parse_args()
+
+    # Check all inputs are valid
+    if isinstance(args.seed, int):
+        random.seed(args.seed)
+    elif args.seed is None:
+        pass
+    else:
+        print("Seed must be an integer.")
+        sys.exit(0)
+
+    if isinstance(args.width, int) and args.width>0:
+        w = args.width
+    elif args.width is None:
+        w = 15
+    else:
+        print("Width must be a positive integer.")
+        sys.exit(0)
+
+    if isinstance(args.height, int) and args.height>0:
+        h = args.height
+    elif args.height is None:
+        h = 15
+    else:
+        print("Height must be a positive integer.")
+        sys.exit(0)
+
+    if isinstance(args.poi, int) and (args.poi >= 0) and (args.poi <= w*h):
+        num_poi = args.poi
+    elif (args.poi is None) and (3 <= w*h):
+        num_poi = 3
+    elif (args.poi is None) and (3 > w*h):
+        print("Too many POIs for grid size.")
+        sys.exit(0)
+    else:
+        print("Number of POIs must be an integer and between 0 and total size of grid.")
+        sys.exit(0)
+
+    if isinstance(args.chargers, int) and (args.chargers >= 0) and (args.chargers <= w*h):
+        num_cs = args.chargers
+    elif (args.chargers is None) and (4 <= w*h):
+        num_cs = 4
+    elif (args.chargers is None) and (4 > w*h):
+        print("Too many chargers for grid size.")
+        sys.exit(0)
+    else:
+        print("Number of chargers must be an integer and between 0 and total size of grid.")
+        sys.exit(0)
+
+    if isinstance(args.new_chargers, int) and (args.new_chargers >= 0) and (args.new_chargers <= w*h - num_cs):
+        num_new_cs = args.new_chargers
+    elif (args.new_chargers is None) and (2 <= w*h - num_cs):
+        num_new_cs = 2
+    elif (args.new_chargers is None) and (2 > w*h - num_cs):
+        print("Too many chargers for grid size and existing charger count.")
+        sys.exit(0)
+    else:
+        print("Number of chargers must be an integer and between 0 and total size of grid minus existing chargers.")
+        sys.exit(0)
+    
+    return w, h, num_poi, num_cs, num_new_cs
```

Comment:
Another option is to return `args`, after all the checks. That reduces potential confusion and simplifies code update (e.g. is/was `num_cs` 4th or 5th element of this tuple?)

Reply:
Can you explain what you mean by this?

Reply:
I think what @randomir means is that if you do `return args`, then when you call this function, you can refer to each argument as `args.width`, `args.height`, etc, rather than returning each arg as an individual variable. 

Does this make sense?